src/lib/pickles/wrap_verifier.ml

```diff
@@ -626,8 +808,218 @@ struct
+                    ()
+                | Maybe (b, z) ->
+                    let z = Array.map z ~f:(fun z -> (b, z)) in
+                    absorb sponge Without_degree_bound z
+                | Some z ->
+                    let z = Array.map z ~f:(fun z -> (Boolean.true_, z)) in
+                    absorb sponge Without_degree_bound z
+              in
+              match[@warning "-4"]
+                (m.lookup_table_comm, m.runtime_tables_selector)
+              with
+              | _ :: Some _ :: _, _ | _, Some _ ->
+                  let joint_combiner = sample_scalar () in
+                  absorb_sorted_first_part () ;
+                  absorb_sorted_second_part () ;
+                  Types.Opt.Some joint_combiner
+              | _ :: None :: _, None ->
+                  absorb_sorted_first_part () ;
+                  absorb_sorted_second_part () ;
+                  Types.Opt.Some { inner = Field.zero }
+              | _ ->
+                  failwith "TODO" )
+        in
+        let lookup_table_comm =
+          match (messages.lookup, joint_combiner) with
+          | Types.Opt.None, Types.Opt.None ->
+              Types.Opt.None
+          | ( Types.Opt.Maybe (b_l, l)
+            , Types.Opt.Maybe (b_joint_combiner, joint_combiner) ) ->
+              ignore ((b_l, l, b_joint_combiner, joint_combiner) : _ * _ * _ * _) ;
+              failwith "TODO"
+          | Types.Opt.Some l, Types.Opt.Some joint_combiner ->
+              let (first_column :: second_column :: rest) =
+                Vector.map
+                  ~f:(Types.Opt.map ~f:(fun x -> [| x |]))
+                  m.lookup_table_comm
+              in
+              let second_column_with_runtime =
+                match (second_column, l.runtime) with
+                | Types.Opt.None, comm | comm, Types.Opt.None ->
+                    comm
+                | ( Types.Opt.Maybe (has_second_column, second_column)
+                  , Types.Opt.Maybe (has_runtime, runtime) ) ->
+                    let second_with_runtime =
+                      let sum =
+                        Array.map2_exn ~f:Inner_curve.( + ) second_column
+                          runtime
+                      in
+                      Array.map2_exn second_column sum
+                        ~f:(fun second_column sum ->
+                          Inner_curve.if_ has_runtime ~then_:sum
+                            ~else_:second_column )
+                    in
+                    let res =
+                      Array.map2_exn second_with_runtime runtime
+                        ~f:(fun second_with_runtime runtime ->
+                          Inner_curve.if_ has_second_column
+                            ~then_:second_with_runtime ~else_:runtime )
+                    in
+                    let b = Boolean.(has_second_column ||| has_runtime) in
+                    Types.Opt.Maybe (b, res)
+                | ( Types.Opt.Maybe (has_second_column, second_column)
```

Comment:
This case and the following case could be merged I think.

Reply:
They can't; the logic is different.

Reply:
Isn't just about naming the variable the same way?